### DISCUSSION TITLE
npm installation gets stuck at reify:mathjs

### DISCUSSION THREAD

**Discussion by Yersi88** [mKeRix/room-assistant#1142] [Q&A]
![image](https://user-images.githubusercontent.com/77145281/162621129-c4223948-9279-4ad3-a775-600ca6f2925e.png)

Not sure how to fix this? It always gets stuck on there. I'm trying to install on rpi zero 2 w

**Comment by Yersi88** [comment]
![image](https://user-images.githubusercontent.com/77145281/162623594-cd5c6784-f048-4041-bf50-e5db66beb8b4.png)
Or this error after I clean the cache with `npm cache clean --force`

**Comment by LazyDave76** [comment]
The version of nodejs that is installed is too new, try using 14.19.3 instead.

https://nodejs.org/download/release/v14.19.3/

**Reply by g4n-g4n** [reply]
Is this the best version of node js to use?

**Comment by Duke-Box** [comment]
Hi,

I also had this issue when installing on a Pi zero W.
It appeared to stop at the mathjs line.
I had several attempts and eventually used the verbose command line.

sudo npm i --global --unsafe-perm --verbose --foreground-scripts room-assistant

If you are patient, you can see it fallback to build. And then if you are really patient you can see it rebuilding the component and then the installation succeeds.

On my pi zero, this took 47 mins from start of the RA installation to getting the success banners.
Hope this helps
